src/test/java/org/jenkinsci/plugins/envinject/EnvInjectEnvVarsJobTest.java

```diff
@@ -0,0 +1,33 @@
+package org.jenkinsci.plugins.envinject;
+
+import hudson.EnvVars;
+import hudson.model.FreeStyleProject;
+import hudson.model.TaskListener;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import static org.junit.Assert.assertNotNull;
+
+public class EnvInjectEnvVarsJobTest {
+
+    @Rule
+    public JenkinsRule jenkins = new JenkinsRule();
+
+    @Test
+    public void EnvVarsJob() throws Exception {
+        FreeStyleProject project = jenkins.createFreeStyleProject();
+
+        EnvInjectJobPropertyInfo jobPropertyInfo = new EnvInjectJobPropertyInfo(null, "REPO=trivial-maven", null, null, null, false);
+        EnvInjectJobProperty envInjectJobProperty = new EnvInjectJobProperty();
+        envInjectJobProperty.setOn(true);
+        envInjectJobProperty.setInfo(jobPropertyInfo);
+        project.addProperty(envInjectJobProperty);
+
+        TaskListener listener = jenkins.createTaskListener();
+        EnvVars environment = project.getEnvironment(jenkins.getInstance(), listener);
+
+        assertNotNull(environment.get("REPO"));
+        jenkins.assertBuildStatusSuccess(project.scheduleBuild2(0).get());
```

Comment:
Running the build does not really accomplish anything. You could delete this line, or make the test demonstrate that the variable is also accessible to build steps. Which reminds me that it is likely that other code in this plugin is interpreting `EnvInjectJobProperty` on a `Run` (or `AbstractBuild`) that could probably now be deleted, since `Run` environments inherit from `Job` environments.